dataprocspawner/spawner.py

```diff
@@ -1024,6 +1027,28 @@ def get_rand_string(self, length):
     rand_str = ''.join((random.choice(letters_and_digits) for i in range(length)))
     return rand_str
 
+  def _validate_zones(self, region, zones):
+    '''
+      Checks if 'dataproc_locations_list' contains zones, relevant for the current region.
+      If variable is not defined or contains irrelevant zones only, then the user is free to use
+      all zones present in current region. Otherwise returns relevant zones only.
+    '''
+    command = f'gcloud compute zones list --filter="name~{region}" --format="value(name)"'
```

Comment:
Is there a reason to use gcloud? I believe we have the compute client libraries available in the Docker container, and that would avoid some interprocess overhead.

Reply:
To follow up on this: this actually isn't very feasible because for unknown reasons, the client libraries don't support filtering on regexs, but only exact matches.

The overhead and latency of gcloud should be somewhat mitigated by the change above of only running this once for each user.

Reply:
Turns out this is possible with the following syntax for regex matches:
` compute.zones().list(project='dingj-knox', filter='name eq .*(us-central1).*').execute()`

We should use the client libraries.

Reply:
This filter also works region= "https://www.googleapis.com/compute/v1/projects/<PROJECT>/regions/<REGION>"

Reply:
GCloud SDK has been replaced with Python libraries. I used `filter='name eq .*(us-central1).*'`